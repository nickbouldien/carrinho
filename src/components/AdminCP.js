import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Bar, BarChart, CartesianGrid, Legend, Tooltip, YAxis, XAxis } from 'recharts';
import { Container, Header } from 'semantic-ui-react';
// import AddItemForm from '../components/AddItemForm';

class AdminCP extends Component {
  componentDidMount() {
    this.props.getItemsInStock();
  }
  render() {
    const { stock } = this.props;
    return (
      <Container>
        {/* <AddItemForm /> */}
        <Header as="h2">Store stock: </Header>

        {
          stock && stock.length ? (
            <BarChart width={730} height={250} data={stock}>
              <CartesianGrid strokeDasharray="3 3" />
              <XAxis dataKey="name" />
              <YAxis />
              <Tooltip />
              <Legend />
              <Bar dataKey="stock" fill="#8884d8" />
            </BarChart>
          ) : (
            <p>No items in stock.</p>
          )
        }

      </Container>
    )
  }
}

AdminCP.propTypes = {
  stock: PropTypes.array.isRequired,
};

export default AdminCP;
