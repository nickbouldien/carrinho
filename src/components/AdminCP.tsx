import React from 'react';
import {Helmet} from 'react-helmet';
import {Bar, BarChart, CartesianGrid, Legend, Tooltip, YAxis, XAxis} from 'recharts';
import {Container, Header} from 'semantic-ui-react';
import {useAppState} from '../providers/AppProvider';
import {IItem} from './ItemCard';
import AddItemForm from '../components/AddItemForm';

const AdminCP: React.FC = () => {
  let stock: IItem[] = [];

  let {
    state: {stockItems},
  } = useAppState();

  if (stockItems && Object.keys(stockItems).length > 0) {
    stock = Object.values(stockItems);
  }

  return (
    <Container>
      <Helmet>
        <title>Carrinho - admin</title>
      </Helmet>

      <AddItemForm />

      <Header as="h2">Store stock: </Header>

      {stock.length > 0 ? (
        <BarChart width={730} height={250} data={stock}>
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="name" />
          <YAxis />
          <Tooltip />
          <Legend />
          <Bar dataKey="stock" fill="#8884d8" />
        </BarChart>
      ) : (
        <p>No items in stock.</p>
      )}
    </Container>
  );
};

export default AdminCP;
