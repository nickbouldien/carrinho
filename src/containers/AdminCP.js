import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { BarChart, YAxis, XAxis, Tooltip, Legend, CartesianGrid, Bar } from 'recharts';
import { Container, Header, Divider /*, Grid, Button, Flag, Dimmer, Loader*/ } from 'semantic-ui-react';
import AddItemFrom from '../components/AddItemForm';

class AdminCP extends Component {

  componentDidMount() {
    // call to fetch stockItems for redux state
    this.props.getItemsInStock();
  }

  render() {
    const { stock } = this.props;
    const chartData = stock;

    return (
      <Container>
  
        <AddItemFrom />
  
        <Divider />

        <Header as="h2">Store stock: </Header>

        <BarChart width={730} height={250} data={chartData}>
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="name" />
          <YAxis />
          <Tooltip />
          <Legend />
          <Bar dataKey="stock" fill="#8884d8" />
        </BarChart>
  
      </Container>
    )
  }
}

AdminCP.propTypes = {
  stock: PropTypes.array.isRequired,
};

const mapStateToProps = (state /*, ownProps*/) => ({
  stock: state.storeStock.stockItems
});

const mapDispatchToProps = (dispatch) => ({
  getItemsInStock: () => {
    dispatch({ type: 'FETCH_CART_ITEMS' });
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(AdminCP);


