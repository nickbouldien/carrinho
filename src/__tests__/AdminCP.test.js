import React from 'react';
import { cleanup, fireEvent, render } from 'react-testing-library';

import AdminCP from '../components/AdminCP';

afterEach(cleanup);

describe("tests for ItemCardInput component", () => {
  const mockFn = jest.fn();
  const emptyStock = [];
  const stockItems = [
    {
      id: "id-1234",
      stock: 172,
      name: "fake item name",
    },
    {
      id: "id-5678",
      stock: 812,
      name: "mock item 2",
    }
  ];

  const { container, getByText, queryByText } = render(<AdminCP stock={emptyStock} getItemsInStock={mockFn} />);
  const h2 = getByText("Store stock:");
  const p = queryByText("No items in stock.");

  test("getItemsInStock prop function is called", () => {
    expect(mockFn).toHaveBeenCalledTimes(1);
    expect(mockFn).toHaveBeenCalledWith();
  });

  test("h2 header is shown", () => {
    expect(h2.textContent).toBe("Store stock: ");
  });

  test("displays 'No items in stock.' when there are no stock items", () => {
    expect(p.textContent).toBe("No items in stock.");
  });

  test("displays bar chart with items when they exist as props", () => {
    const { container } = render(<AdminCP stock={stockItems} getItemsInStock={mockFn} />);
    const tooltip = container.querySelectorAll('recharts-tooltip-label');
    expect(tooltip.length).toBe(0);

    const bars = container.querySelectorAll('.recharts-bar-rectangle');
    expect(bars.length).toBe(stockItems.length);
  });

  test.skip("displays (correct) tooltip when hovering over a bar on bar chart", () => {
    // TODO
  });

  test("snapshot test with data (stock)", () => {
    const { container } = render(<AdminCP stock={stockItems} getItemsInStock={mockFn} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("snapshot test without data (no stock)", () => {
    const { container } = render(<AdminCP stock={emptyStock} getItemsInStock={mockFn} />);
    expect(container.firstChild).toMatchSnapshot();
  });

});
