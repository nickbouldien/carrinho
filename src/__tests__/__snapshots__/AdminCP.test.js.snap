// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for ItemCardInput component snapshot test with data (stock) 1`] = `
<div
  class="ui container"
>
  <h2
    class="ui header"
  >
    Store stock: 
  </h2>
  <div
    class="recharts-wrapper"
    style="position: relative; cursor: default; width: 730px; height: 250px;"
  >
    <svg
      class="recharts-surface"
      width="730"
      height="250"
      viewBox="0 0 730 250"
      version="1.1"
    >
      <g
        class="recharts-cartesian-grid"
      >
        <g
          class="recharts-cartesian-grid-horizontal"
        >
          <line
            width="660"
            stroke-dasharray="3 3"
            fill="none"
            x="65"
            y="5"
            stroke="#ccc"
            height="210"
            x1="65"
            y1="215"
            x2="725"
            y2="215"
          />
          <line
            width="660"
            stroke-dasharray="3 3"
            fill="none"
            x="65"
            y="5"
            stroke="#ccc"
            height="210"
            x1="65"
            y1="162.5"
            x2="725"
            y2="162.5"
          />
          <line
            width="660"
            stroke-dasharray="3 3"
            fill="none"
            x="65"
            y="5"
            stroke="#ccc"
            height="210"
            x1="65"
            y1="110"
            x2="725"
            y2="110"
          />
          <line
            width="660"
            stroke-dasharray="3 3"
            fill="none"
            x="65"
            y="5"
            stroke="#ccc"
            height="210"
            x1="65"
            y1="57.5"
            x2="725"
            y2="57.5"
          />
          <line
            width="660"
            stroke-dasharray="3 3"
            fill="none"
            x="65"
            y="5"
            stroke="#ccc"
            height="210"
            x1="65"
            y1="5"
            x2="725"
            y2="5"
          />
        </g>
        <g
          class="recharts-cartesian-grid-vertical"
        >
          <line
            width="660"
            stroke-dasharray="3 3"
            fill="none"
            x="65"
            y="5"
            stroke="#ccc"
            height="210"
            x1="230"
            y1="5"
            x2="230"
            y2="215"
          />
          <line
            width="660"
            stroke-dasharray="3 3"
            fill="none"
            x="65"
            y="5"
            stroke="#ccc"
            height="210"
            x1="560"
            y1="5"
            x2="560"
            y2="215"
          />
          <line
            width="660"
            stroke-dasharray="3 3"
            fill="none"
            x="65"
            y="5"
            stroke="#ccc"
            height="210"
            x1="65"
            y1="5"
            x2="65"
            y2="215"
          />
          <line
            width="660"
            stroke-dasharray="3 3"
            fill="none"
            x="65"
            y="5"
            stroke="#ccc"
            height="210"
            x1="725"
            y1="5"
            x2="725"
            y2="215"
          />
        </g>
      </g>
      <g
        class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
      >
        <line
          stroke="#666"
          class="recharts-cartesian-axis-line"
          height="30"
          x="65"
          y="215"
          width="660"
          fill="none"
          x1="65"
          y1="215"
          x2="725"
          y2="215"
        />
        <g
          class="recharts-cartesian-axis-ticks"
        >
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              stroke="#666"
              class="recharts-cartesian-axis-tick-line"
              height="30"
              x="65"
              y="215"
              width="660"
              fill="none"
              x1="230"
              y1="221"
              x2="230"
              y2="215"
            />
            <text
              width="660"
              height="30"
              x="230"
              y="223"
              stroke="none"
              fill="#666"
              class="recharts-text recharts-cartesian-axis-tick-value"
              text-anchor="middle"
            >
              <tspan
                x="230"
                dy="0.71em"
              >
                fake item name
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              stroke="#666"
              class="recharts-cartesian-axis-tick-line"
              height="30"
              x="65"
              y="215"
              width="660"
              fill="none"
              x1="560"
              y1="221"
              x2="560"
              y2="215"
            />
            <text
              width="660"
              height="30"
              x="560"
              y="223"
              stroke="none"
              fill="#666"
              class="recharts-text recharts-cartesian-axis-tick-value"
              text-anchor="middle"
            >
              <tspan
                x="560"
                dy="0.71em"
              >
                mock item 2
              </tspan>
            </text>
          </g>
        </g>
      </g>
      <g
        class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
      >
        <line
          stroke="#666"
          class="recharts-cartesian-axis-line"
          height="210"
          x="5"
          y="5"
          width="60"
          fill="none"
          x1="65"
          y1="5"
          x2="65"
          y2="215"
        />
        <g
          class="recharts-cartesian-axis-ticks"
        >
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              stroke="#666"
              class="recharts-cartesian-axis-tick-line"
              height="210"
              x="5"
              y="5"
              width="60"
              fill="none"
              x1="59"
              y1="215"
              x2="65"
              y2="215"
            />
            <text
              width="60"
              height="210"
              x="57"
              y="215"
              stroke="none"
              fill="#666"
              class="recharts-text recharts-cartesian-axis-tick-value"
              text-anchor="end"
            >
              <tspan
                x="57"
                dy="0.355em"
              >
                0
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              stroke="#666"
              class="recharts-cartesian-axis-tick-line"
              height="210"
              x="5"
              y="5"
              width="60"
              fill="none"
              x1="59"
              y1="162.5"
              x2="65"
              y2="162.5"
            />
            <text
              width="60"
              height="210"
              x="57"
              y="162.5"
              stroke="none"
              fill="#666"
              class="recharts-text recharts-cartesian-axis-tick-value"
              text-anchor="end"
            >
              <tspan
                x="57"
                dy="0.355em"
              >
                250
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              stroke="#666"
              class="recharts-cartesian-axis-tick-line"
              height="210"
              x="5"
              y="5"
              width="60"
              fill="none"
              x1="59"
              y1="110"
              x2="65"
              y2="110"
            />
            <text
              width="60"
              height="210"
              x="57"
              y="110"
              stroke="none"
              fill="#666"
              class="recharts-text recharts-cartesian-axis-tick-value"
              text-anchor="end"
            >
              <tspan
                x="57"
                dy="0.355em"
              >
                500
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              stroke="#666"
              class="recharts-cartesian-axis-tick-line"
              height="210"
              x="5"
              y="5"
              width="60"
              fill="none"
              x1="59"
              y1="57.5"
              x2="65"
              y2="57.5"
            />
            <text
              width="60"
              height="210"
              x="57"
              y="57.5"
              stroke="none"
              fill="#666"
              class="recharts-text recharts-cartesian-axis-tick-value"
              text-anchor="end"
            >
              <tspan
                x="57"
                dy="0.355em"
              >
                750
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              stroke="#666"
              class="recharts-cartesian-axis-tick-line"
              height="210"
              x="5"
              y="5"
              width="60"
              fill="none"
              x1="59"
              y1="5"
              x2="65"
              y2="5"
            />
            <text
              width="60"
              height="210"
              x="57"
              y="5"
              stroke="none"
              fill="#666"
              class="recharts-text recharts-cartesian-axis-tick-value"
              text-anchor="end"
            >
              <tspan
                x="57"
                dy="0.355em"
              >
                1000
              </tspan>
            </text>
          </g>
        </g>
      </g>
      <g
        class="recharts-layer recharts-bar"
      >
        <g
          class="recharts-layer recharts-bar-rectangles"
        >
          <g
            class="recharts-layer"
          >
            <g
              class="recharts-layer recharts-bar-rectangle"
            />
            <g
              class="recharts-layer recharts-bar-rectangle"
            />
          </g>
        </g>
      </g>
    </svg>
    <div
      class="recharts-legend-wrapper"
      style="position: absolute; width: 720px; height: auto; left: 5px; bottom: 5px;"
    >
      <ul
        class="recharts-default-legend"
        style="padding: 0px; margin: 0px; text-align: center;"
      >
        <li
          class="recharts-legend-item legend-item-0"
          style="display: inline-block; margin-right: 10px;"
        >
          <svg
            class="recharts-surface"
            width="14"
            height="14"
            style="display: inline-block; vertical-align: middle; margin-right: 4px;"
            viewBox="0 0 32 32"
            version="1.1"
          >
            <path
              stroke="none"
              fill="#8884d8"
              d="M0,4h32v24h-32z"
              class="recharts-legend-icon"
            />
          </svg>
          <span
            class="recharts-legend-item-text"
          >
            stock
          </span>
        </li>
      </ul>
    </div>
    <div
      class="recharts-tooltip-wrapper"
      style="pointer-events: none; visibility: hidden; position: absolute; top: 0px;"
    >
      <div
        class="recharts-default-tooltip"
        style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
      >
        <p
          class="recharts-tooltip-label"
          style="margin: 0px;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`tests for ItemCardInput component snapshot test without data (no stock) 1`] = `
<div
  class="ui container"
>
  <h2
    class="ui header"
  >
    Store stock: 
  </h2>
  <p>
    No items in stock.
  </p>
</div>
`;
